homeassistant/components/roon/config_flow.py

```diff
@@ -22,39 +23,71 @@
 TIMEOUT = 120
 
 
-class RoonHub:
+class RoonHub:  # pragma: no cover
     """Interact with roon during config flow."""
 
-    def __init__(self, host):
-        """Initialize."""
-        self._host = host
+    def __init__(self, hass):
+        """Initialise the RoonHub."""
+        self._hass = hass
 
-    async def authenticate(self, hass) -> bool:
-        """Test if we can authenticate with the host."""
+    async def discover(self):
+        """Try and discover roon servers."""
+        discovery = RoonDiscovery(None)
+        servers = await self._hass.async_add_executor_job(discovery.all)
+        _LOGGER.debug("Servers = %s", servers)
+        await self._hass.async_add_executor_job(discovery.stop)
```

Comment:
Please do both sync calls in the same function and thread context. It's expensive to switch context.

Reply:
```py
def get_discovered_servers(discovery):
    servers = discovery.all()
    discovery.stop()
    return servers
```